src/services/datalake-storage.ts

```diff
@@ -93,15 +94,15 @@ export default class DataLakeStorage implements StorageService {
     return this.getDirectoryClient(directory).deleteIfExists(true);
   }
 
-  public async listFiles(directory: string): Promise<Record<string, unknown>[]> {
+  public async listFiles(directory: string): Promise<DataLakeFileEntry[]> {
     const files = await this.fileSystemClient.listPaths({ path: directory });
-    const fileList = [];
+    const fileList: DataLakeFileEntry[] = [];
 
     for await (const file of files) {
       if (file.name === undefined) {
         continue;
       }
-      fileList.push({ name: basename(file.name), path: file.name, isDirectory: file.isDirectory });
+      fileList.push({ name: basename(file.name), path: file.name, isDirectory: file.isDirectory ?? false });
```

Comment:
Does `file.isDirectory` sometimes return undefined?

Reply:
According to the type hinting in the class... I thought it was weird to but here we are.  Not our code sadly so we just have to roll with it.